Set-DomainUser.ps1

```diff
@@ -1,720 +1,1124 @@
+    
+    Write-Host "Logging in to CyberArk"
+    $pvwaToken = New-ConnectionToRestAPI -pvwaAddress $pvwaAddress -tinaCreds $tinaCreds
+    if (Test-PvwaToken -Token $pvwaToken -pvwaAddress $pvwaAddress) {
+        # Get platform info
+        Write-Host "Checking current platform status"
+        $platformStatus = Get-PlatformStatus -pvwaAddress $pvwaAddress -pvwaToken $pvwaToken -PlatformId $PlatformName
+        if ($platformStatus -eq $false) {
+            # function returns false if platform does not exist
+            # Creating Platform
+            Write-Host "Creating new platform"
+            Duplicate-Platform -pvwaAddress $pvwaAddress -pvwaToken $pvwaToken -CurrentPlatformId "7" -NewPlatformName $PlatformName -NewPlatformDescription "Platform for PSM accounts"
+            $Tasks += ("Set appropriate policies and settings on platform `"{0}`"" -f $PlatformName)
+            # Get platform info again so we can ensure it's activated
+            $platformStatus = Get-PlatformStatus -pvwaAddress $pvwaAddress -pvwaToken $pvwaToken -PlatformId $PlatformName
+        }
+        else {
+            Write-Warning ('Platform {0} already exists. Please verify it meets requirements.' -f $PlatformName)
+            $Tasks += ("Verify that the existing platform `"{0}`" is configured correctly" -f $PlatformName)
+        }
+        if ($platformStatus.Active -eq $false) {
+            Write-Host "Platform is deactivated. Activating."
+            Activate-Platform -pvwaAddress $pvwaAddress -pvwaToken $pvwaToken -Platform $platformStatus.Id
+        }
+        Write-Host "Checking current safe status"
+        $safeStatus = Get-SafeStatus -pvwaAddress $pvwaAddress -pvwaToken $pvwaToken -SafeName $Safe
+        if ($safeStatus -eq $false) {
+            # function returns false if safe does not exist
+            Write-Host "Safe $Safe does not exist. Please create it or provide a different safe name with the -Safe option"
+            exit 1
+        }
+        If (!($safeStatus.managingCpm)) {
+            # Safe exists but no CPM assigned
+            Write-Warning ("There is no Password Manager (CPM) assigned to safe `"{0}`"" -f $Safe)
+            $Tasks += ("Assign a Password Manager (CPM) to safe `"{0}`"" -f $Safe)
+        }
+        # Giving Permission on the safe if we are using UM, The below will give full permission to vault admins
+        If ($UM) {
+            Write-Host "Granting administrators access to PSM safe"
+            Set-SafePermissionsFull -pvwaAddress $pvwaAddress -pvwaToken $pvwaToken -safe $safe
+        }
+        # Creating PSMConnect, We can now add a safe need as well for the below line if we have multiple domains
+        Write-Host "Onboarding PSMConnect Account"
+        $OnboardResult = New-VaultAdminObject -pvwaAddress $pvwaAddress -pvwaToken $pvwaToken -name $PSMConnectAccountName -domain $domain -Credentials $psmConnectCredentials -platformID $PlatformName -safe $safe
```

Comment:
Capture the result of account onboarding so we can catch and handle errors, such as if the account already exists.